gen/topdesk

```diff
@@ -0,0 +1,95 @@
+our $STATUS_VALID;                *STATUS_VALID =            \'VALID';
+
+
+our $userStruc =        {};
+
+my $fileUsers =         $DIRECTORY . $SERVICE_NAME ".csv";
+
+my @resourcesData = $data->getChildElements;
+foreach my $rData (@resourcesData) {
+        my @membersData = $rData->getChildElements;
+        foreach my $mData (@membersData) {
+                processUsers $mData;
+        }
+}
+
+# USERS: first_name last_name login e-mail eppns
+open FILE_USERS,">$fileUsers" or die "Cannot open $fileUsers: $! \n";
+binmode FILE_USERS, ":utf8";
+
+# Print CSV header
+print FILE_USERS "Surname;First Name(s);Initials;Prefixes;Title;Gender;Language;Site;Location;Budget holder;Job Title;Employee Number;Department;Telephone;Mobile Number;Fax Number;Email;Network Login Name;Mainframe;Self Service Desk;Has Attention;Attention;Comments\n";
+
+# Example of one line:
+# Prochazka;Michal;MP;;;;;ELIXIR-AAI-USER;;;;;;;;;michalp@ics.muni.cz;abcd@elixir-europe.org;abcd@elixir-europe.org;abcd@elixir-europe.org;;;ELIXIR
+
+foreach my $login (sort keys %$userStruc) {
+
+	if ($userStruc->{$login}->{$A_USER_LASTNAME}) {
+                print FILE_USERS "$userStruc->{$login}->{$A_USER_LASTNAME}";
+        }
+	print FILE_USERS ";";
+        if ($userStruc->{$login}->{$A_USER_FIRSTNAME}) {
+                print FILE_USERS "$userStruc->{$login}->{$A_USER_FIRSTNAME}";
+        }
+	print FILE_USERS ";";
+        if ($userStruc->{$login}->{$A_USER_LASTNAME} && $userStruc->{$login}->{$A_USER_FIRSTNAME}) {
+                print FILE_USERS substr($userStruc->{$login}->{$A_USER_FIRSTNAME}, 0, 1) . substr($userStruc->{$login}->{$A_USER_LASTNAME}, 0, 1);
+        }
+	print FILE_USERS ";";
+        print FILE_USERS ";;;;ELIXIR-AAI-USER;;;;;;;;;";
+        print FILE_USERS "$userStruc->{$login}->{$A_USER_EMAIL};";
+        print FILE_USERS "$login;$login;$login;";
+        print FILE_USERS ";;ELIXIR\n";
+}
+
+close (FILE_USERS) or die "Cannot close $fileUsers: $! \n";
+
+perunServicesInit::finalize;
+
+
+##############################################################################
+#   Only subs definitions down there
+##############################################################################
+
+## creates structure for users.csv file
+sub processUsers {
+        my ($memberData) = @_;
+
+        my %memberAttributes = attributesToHash $memberData->getAttributes;
+
+        if ($memberAttributes{$A_USER_STATUS} eq $STATUS_VALID) {
+                my $firstName =        $memberAttributes{$A_USER_FIRSTNAME};
+                my $lastName =         $memberAttributes{$A_USER_LASTNAME};
+                my $login =            $memberAttributes{$A_USER_LOGIN};
+                my $email =            $memberAttributes{$A_USER_EMAIL};
+
+                $userStruc->{$login}->{$A_USER_FIRSTNAME} =  $firstName;
+                $userStruc->{$login}->{$A_USER_LASTNAME} =   $lastName;
+                $userStruc->{$login}->{$A_USER_EMAIL} =      $email;
+        }
```

Comment:
If you don't mind having more data in userStruc structure you can squeeze whole "if" statement to following:
```perl
$userStruc->{$memberAttributes{$A_USER_LOGIN}} = \%memberAttributes;
```

It's only suggestion. Fix it or not, as you like.